app/javascript/shared/controllers/content_list_controller.js

```diff
@@ -0,0 +1,38 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+  showCount = 10;
+  step = 10;
+  items = [];
+  buttonEl = null;
+
+  connect() {
+    this.buttonEl = this.element.getElementsByClassName('load-more-btn')[0];
+    this.items = this.element.getElementsByClassName('content-item');
+    this.checkButton();
+    for (let i = this.showCount; i < this.items.length; i += 1) {
+      this.items[i].style.display = 'none';
+    }
+    $(this.buttonEl).click(this.loadMore.bind(this));
+  }
+
+  loadMore() {
+    this.showCount += this.step;
+
+    for (let i = 0; i < this.items.length; i += 1) {
+      this.items[i].style.display = this.showCount <= i ? 'none' : 'block';
```

Comment:
hmm, would be better to just load first 10 from the server and then load more, but let's have it in the next version.